financialaid/views.py

```diff
@@ -234,3 +227,48 @@ class FinancialAidActionView(UpdateAPIView):
     lookup_field = "id"
     lookup_url_kwarg = "financial_aid_id"
     queryset = FinancialAid.objects.all()
+
+
+class GetLearnerPriceForCourseView(APIView):
+    """
+    View for retrieving a leaner's price for a course run
+    """
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, UserCanViewLearnerCoursePrice)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get request for GetLearnerPriceForCourseView
+        """
+        learner = get_object_or_404(User, id=self.kwargs["user_id"])
+        program = get_object_or_404(Program, id=self.kwargs["program_id"], live=True)
+        self.check_object_permissions(request, {"learner": learner, "program": program})
+        # Validate that learner is enrolled in program
+        try:
+            ProgramEnrollment.objects.get(user=learner, program=program)
+        except ProgramEnrollment.DoesNotExist:
+            raise ValidationError("Learner not enrolled in this program.")
+        course_price = program.get_course_price()
+        has_financial_aid_request = False
+        financial_aid_adjustment = False
+        try:
+            # Check to see if learner has a financial aid request
+            financial_aid = FinancialAid.objects.get(
+                user=learner,
+                tier_program__program=program,
+                tier_program__program__financial_aid_availability=True
+            )
+            has_financial_aid_request = True
+            if financial_aid.status == FinancialAidStatus.APPROVED:
+                # If the financial aid request is approved, adjust course price
+                course_price = course_price - financial_aid.tier_program.discount_amount
+                financial_aid_adjustment = True
```

Comment:
That would make it a less clear in terms of readibility.  Doing so would require initializing `financial_aid = None` first in case it doesn't exist and also changing `if financial_aid.status == FinancialAidStatus.APPROVED:` to `if financial_aid is not None and financial_aid.status == FinancialAidStatus.APPROVED`.

I personally think this adds unnecessary bulk, but if you disagree then I'm happy to change it.